pkg/pipeline/sink/uploader/s3.go

```diff
@@ -34,18 +37,42 @@ const (
 	getBucketLocationRegion = "us-east-1"
 )
 
+// CustomRetryer wraps the SDK's built in DefaultRetryer adding additional
+// custom features. Namely, to always retry.
+type CustomRetryer struct {
+	client.DefaultRetryer
+}
+
+// ShouldRetry overrides the SDK's built in DefaultRetryer because the PUTs for segments/playlists are always idempotent
+func (r CustomRetryer) ShouldRetry(req *request.Request) bool {
+	return true
```

Comment:
I think this still needs some safeguards - in the case of bad credentials (which happens fairly often in cloud), it will take much longer to fail

Reply:
added check for HTTP status codes 401 or 403 and does not retry for those